mobile/android/android-device-file-system.ts

```diff
@@ -85,18 +86,18 @@ export class AndroidDeviceFileSystem implements Mobile.IDeviceFileSystem {
 		const devicePaths: string[] = [];
 		const currentShasums: IStringDictionary = {};
 
-		await Promise.all(
-			localToDevicePaths.map(async localToDevicePathData => {
-				const localPath = localToDevicePathData.getLocalPath();
-				const stats = this.$fs.getFsStats(localPath);
-				if (stats.isFile()) {
-					const fileShasum = await this.$fs.getFileShasum(localPath);
-					currentShasums[localPath] = fileShasum;
-				}
+		const action = async (localToDevicePathData: Mobile.ILocalToDevicePathData) => {
+			const localPath = localToDevicePathData.getLocalPath();
+			const stats = this.$fs.getFsStats(localPath);
+			if (stats.isFile()) {
```

Comment:
this code seems like a duplicate of part of the code from `updateHashes` in android-device-hash-service. Consider extracting a new method that executes the common part.